File: tests_generated/test_use_github_actions-False_deployment_language-typescript/smart_contracts/deploy-config.ts
import * as algokit from '@algorandfoundation/algokit-utils'
import { AppSpec } from '@algorandfoundation/algokit-utils/types/app-spec'
import { HelloWorldAppClient } from './artifacts/hello_world_app/client'

// Edit this to add in your contracts
export const contracts = ['HelloWorldApp'] as const

export async function deploy(name: (typeof contracts)[number], appSpec: AppSpec) {
  const algod = algokit.getAlgoClient()
  const indexer = algokit.getAlgoIndexerClient()
  const deployer = await algokit.getAccount({ name: 'DEPLOYER', fundWith: algokit.algos(3) }, algod)
  await algokit.ensureFunded(
    {
      accountToFund: deployer,
      minSpendingBalance: algokit.algos(2),
      minFundingIncrement: algokit.algos(2),
    },
    algod,
  )
  const isMainNet = await algokit.isMainNet(algod)
  const appClient = new HelloWorldAppClient(
    {
      resolveBy: 'creatorAndName',
      findExistingUsing: indexer,
      sender: deployer,
      creatorAddress: deployer.addr,
    },
    algod,
  )

  switch (name) {
    // Edit this to add the custom deployment logic for each contract
    case 'HelloWorldApp':
      const app = await appClient.deploy({
        allowDelete: !isMainNet,
        allowUpdate: !isMainNet,
        onSchemaBreak: isMainNet ? 'append' : 'replace',
        onUpdate: isMainNet ? 'append' : 'update',
      })
      // If app was just created fund the app account
      if (['create', 'replace'].includes(app.operationPerformed)) {
        algokit.transferAlgos(
          {
            amount: algokit.algos(1),
            from: deployer,
            to: app.appAddress,
          },
          algod,
        )
      }

      const method = 'hello'
      const response = await appClient.hello({ name: 'world' })
      console.log(`Called ${method} on ${name} (${app.appId}) with name = world, received: ${response.return}`)
      break
    default:
      throw new Error(`Attempt to deploy unknown contract ${name}`)
  }
}

File: tests_generated/generators/test_smart_contract_generator_default_production_preset_python/smart_contracts/hello_world/deploy_config.py
import logging

import algokit_utils
from algosdk.v2client.algod import AlgodClient
from algosdk.v2client.indexer import IndexerClient

logger = logging.getLogger(__name__)


# define deployment behaviour based on supplied app spec
def deploy(
    algod_client: AlgodClient,
    indexer_client: IndexerClient,
    app_spec: algokit_utils.ApplicationSpecification,
    deployer: algokit_utils.Account,
) -> None:
    from smart_contracts.artifacts.hello_world.client import (
        HelloWorldClient,
    )

    app_client = HelloWorldClient(
        algod_client,
        creator=deployer,
        indexer_client=indexer_client,
    )
    is_mainnet = algokit_utils.is_mainnet(algod_client)
    app_client.deploy(
        on_schema_break=(
            algokit_utils.OnSchemaBreak.AppendApp
            if is_mainnet
            else algokit_utils.OnSchemaBreak.ReplaceApp
        ),
        on_update=algokit_utils.OnUpdate.AppendApp
        if is_mainnet
        else algokit_utils.OnUpdate.UpdateApp,
        allow_delete=not is_mainnet,
        allow_update=not is_mainnet,
    )

    name = "world"
    response = app_client.hello(name=name)
    logger.info(
        f"Called hello on {app_spec.contract.name} ({app_client.app_id}) "
        f"with name={name}, received: {response.return_value}"
    )
